java/memory/memory-core/src/main/java/org/apache/arrow/memory/BoundsChecking.java

```diff
@@ -34,14 +34,15 @@
   static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(BoundsChecking.class);
 
   static {
-    String envProperty = System.getenv("ARROW_ENABLE_UNSAFE_MEMORY_ACCESS");
-    String oldProperty = System.getProperty("drill.enable_unsafe_memory_access");
+    String envProperty = System.getenv("SIREN_ARROW_ENABLE_UNSAFE_MEMORY_ACCESS");
+    String oldProperty = System.getProperty("siren.drill.enable_unsafe_memory_access");
```

Comment:
With this change, which change the arrow default from `false` (originally) to `true` (in Federate, we want to enable unsafe access by default), we can clarify what we want:

```suggestion
    String envProperty = System.getenv().getOrDefault("SIREN_ARROW_ENABLE_UNSAFE_MEMORY_ACCESS", "true");
    String oldProperty = System.getProperty("siren.drill.enable_unsafe_memory_access", "true");
```

```suggestion
     String newProperty = System.getProperty("siren.arrow.enable_unsafe_memory_access", "true`);
```

Then line 58  you can revert to the upstream version, i.e., `BOUNDS_CHECKING_ENABLED = !"true".equals(unsafeFlagValue);`

Reply:
Done here https://github.com/sirensolutions/arrow/pull/16/commits/3aba23a6d40e776d16fb6a1dc90c53138aa90b0e